homeassistant/components/mqtt/__init__.py

```diff
@@ -282,15 +285,33 @@ async def finish_dump(_: datetime) -> None:
     return True
 
 
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
You need to add checks for the config entry version, like this: https://github.com/home-assistant/core/blob/3923aafd18b40356f3449d7825fa2daf6b50999e/homeassistant/components/anova/__init__.py#L76-L95

I think there was some idea to already now prepare for a major version bump, I think that can happen in a follow-up PR to not complicate this PR further.

Reply:
Yes, to support subentries